File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(const char* s) {
    int answer = 0;
    int m = 0;
    int n = 0;
    char mc;
    for (int i = 0; s[i] != '\0'; i++) {
        if (m == 0) {
            mc = s[i];
            m++;
            continue;
        }
        
        if (mc == s[i]) {
            m++;
        }
        else {
            n++;
        }
        
        if (m == n) {
            answer++;
            m = 0;
            n = 0;
        }
    }
    
    if (m > 0) answer++;
    
    return answer;
}

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    answer = 0
    m = 0
    n = 0
    mc = ''
    for i in s:
        if m == 0:
            mc = i
            m += 1
            continue
        
        if mc == i:
            m += 1
        else:
            n += 1
            
        if m == n:
            answer += 1
            m = 0
            n = 0
            
    if m > 0:
        answer += 1
        
    return answer